src/memory_target_ops.h

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#ifndef UMF_MEMORY_TARGET_OPS_H
+#define UMF_MEMORY_TARGET_OPS_H 1
+
+#include <stdbool.h>
+
+#include <umf/base.h>
+#include <umf/memspace.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct umf_memory_target_t *umf_memory_target_handle_t;
+
+struct umf_memory_target_ops_t {
+    /// Version of the ops structure.
+    /// Should be initialized using UMF_VERSION_CURRENT
+    uint32_t version;
+
+    enum umf_result_t (*initialize)(void *params, void **mem_target);
+    void (*finalize)(void *mem_target);
+
+    enum umf_result_t (*pool_create_from_memspace)(
```

Comment:
This is an implementation detail for now.

Reply:
I agree that architecture discussion is needed, but my understanding is the following. Starting with this PR and until we haven't decided to expose it as a public API we have been gaining implementation experience. So let's see how it will glue together (Memspace API on top of low-level pool creation API).